catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/BroadcastBricksTest.java

```diff
@@ -43,98 +43,98 @@
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
 
 import static android.support.test.espresso.Espresso.onView;
 import static android.support.test.espresso.Espresso.pressBack;
 import static android.support.test.espresso.action.ViewActions.clearText;
 import static android.support.test.espresso.action.ViewActions.click;
-import static android.support.test.espresso.action.ViewActions.closeSoftKeyboard;
-import static android.support.test.espresso.action.ViewActions.typeText;
+import static android.support.test.espresso.action.ViewActions.typeTextIntoFocusedView;
 import static android.support.test.espresso.assertion.ViewAssertions.doesNotExist;
 import static android.support.test.espresso.matcher.ViewMatchers.withId;
 import static android.support.test.espresso.matcher.ViewMatchers.withText;
 
 import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.junit.runners.Parameterized.Parameter;
+import static org.junit.runners.Parameterized.Parameters;
 
-@RunWith(AndroidJUnit4.class)
+@Category({Cat.AppUi.class, Level.Smoke.class})
+@RunWith(Parameterized.class)
 public class BroadcastBricksTest {
-	private String defaultMessage = "defaultMessage";
-	private int broadcastSendPosition = 1;
-	private int broadcastReceivePosition = 2;
-
 	@Rule
 	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new
 			BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
 
+	@Parameters(name = "{2}")
+	public static Iterable<Object[]> data() {
+		return Arrays.asList(new Object[][] {
+				{1, R.id.brick_broadcast_spinner, "BroadcastSendTest"},
+				{2, R.id.brick_broadcast_receive_spinner, "BroadcastReceiveTest"},
+				{3, R.id.brick_broadcast_wait_spinner, "BroadcastAndWaitTest"}
+		});
+	}
+
+	@Parameter
+	public int brickPosition;
+
+	@Parameter(1)
+	public @IdRes int spinnerId;
+
+	@Parameter(2)
+	public String testName;
+
+	private String defaultMessage = "defaultMessage";
+
 	@Before
 	public void setUp() throws Exception {
-		createProjectAndGetStartScriptWithImages("BroadcastBrickTest");
+		createProject("BroadcastBricksTest");
 		baseActivityTestRule.launchActivity(null);
 	}
 
-	@Category({Cat.AppUi.class, Level.Smoke.class})
 	@Test
-	public void testBroadcastBricksLayout() {
+	public void removeUnusedMessagesBroadcastTest() {
 		onBrickAtPosition(0).checkShowsText(R.string.brick_when_started);
-		onBrickAtPosition(broadcastSendPosition).checkShowsText(R.string.brick_broadcast);
-		onBrickAtPosition(broadcastReceivePosition).checkShowsText(R.string.brick_broadcast_receive);
-	}
+		onBrickAtPosition(1).checkShowsText(R.string.brick_broadcast);
```

Comment:
please use variable names, it makes it way easier to read than and comprehend than with hardcoded values 1,2,3...